### PR TITLE
#3047-13-tests-for-floating-windows

### PR DESCRIPTION
**Pull request by Vehemen**
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [x] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [ ] PR is linked with the issue
- [ ] base branch (master or release/xx) is correct
- [ ] task status changed to "Code review"

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@svgr/rollup": "8.0.1",
```
Comment (Nitvex) [inline demo/package.json:6]: Why you added dependency to demo project?

File: ketcher-autotests/tests/Templates/Template-Manipulations/Template-manipulations.spec.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+  test('Template palette', async ({ page }) => {
+    /*
+            Test case: 1666
+            Description: Look at the bottom of the application.
+            Choose any template.
+            */
+    await takeEditorScreenshot;
```
Comment (Nitvex) [inline ketcher-autotests/tests/Templates/Template-Manipulations/Template-manipulations.spec.ts:64]: Function is not even invoked – please be attentive

File: ketcher-autotests/tests/Templates/Template-Manipulations/Template-manipulations.spec.ts [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        Open the saved *.mol file and edit it in any way.
+      */
+    await selectRing(RingButton.Benzene, page);
+    const expectedFile = await getMolfile(page, 'v2000');
+    await saveToFile('TM-1737.mol', expectedFile);
+    const METADATA_STRING_INDEX = [1];
+    const { fileExpected: molFileExpected, file: molFile } =
+      await receiveFileComparisonData({
+        page,
+        expectedFileName: 'tests/test-data/template-manipulations/TM-1737.mol',
```
Comment (Nitvex) [inline ketcher-autotests/tests/Templates/Template-Manipulations/Template-manipulations.spec.ts:256]: Could you please give meaningful filenames? `TM-1737.mol` doesn't make any sence

File: ketcher-autotests/tests/User-Interface/Floating windows/floating-windows.spec.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  });
+
+  test('Calculated values: check accuracy', async ({ page }) => {
+    /*
+      Test case: EPMLSOPKET-3999(1)
+      Description: verify 0 decimal places after the dot for calculated values 
+    */
+    await openFileAndAddToCanvas('bicycle.mol', page);
+    await selectTopPanelButton(TopPanelButton.Calculated, page);
+    await page
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Floating windows/floating-windows.spec.ts:41]: This bunch of action is also not very clear for me. Could you add testId for item you want to click? Or create a function for this?

File: packages/ketcher-polymer-editor-react/package.json [hunk 5]
```diff
@@ -62,7 +62,8 @@
     "react-dropzone": "^11.7.1",
     "react-redux": "^7.2.1",
     "redux": "^4.0.5",
-    "redux-saga": "^1.1.3"
+    "redux-saga": "^1.1.3",
+    "serve": "14.2.0"
```
Comment (Nitvex) [inline packages/ketcher-polymer-editor-react/package.json:66]: Please, revert your changes in `package.json`s and `package-lock.json`s files

**Review by Nitvex — CHANGES_REQUESTED** [review]

File: ketcher-autotests/tests/User-Interface/Floating windows/floating-windows.spec.ts (hunk 4 above)
Comment (Vehemen) [reply ketcher-autotests/tests/User-Interface/Floating windows/floating-windows.spec.ts:41]: > This bunch of action is also not very clear for me. Could you add testId for item you want to click? Or create a function for this?

Not hand made. Should I remove that?

File: ketcher-autotests/tests/User-Interface/Floating windows/floating-windows.spec.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+  test('Paste from clipboard/bad data', async ({ page }) => {
+    /*
+      Test case: EPMLSOPKET-4008
+      Description: bad data via paste from clipboard 
+    */
+    await selectTopPanelButton(TopPanelButton.Open, page);
+    await page.getByText('Paste from clipboard').click();
+    await pasteFromClipboardAndAddToCanvas(
+      page,
+      // eslint-disable-next-line max-len
```
[the remark this answers is not in the document]
Comment (Vehemen) [reply ketcher-autotests/tests/User-Interface/Floating windows/floating-windows.spec.ts:139]: Deleted

**Pull request closed without merging**